vip-filesystem/class.vip-filesystem-stream.php

```diff
@@ -0,0 +1,551 @@
+
+		if ( -1 === $result ) {
+			// Seek failed
+			trigger_error( 'Error seeking on file: ' . $this->path,
+				E_USER_WARNING );
+			return FALSE;
+		}
+
+		return TRUE;
+	}
+
+	/**
+	 * Write to a file
+	 *
+	 * @since   1.0.0
+	 * @accesss public
+	 *
+	 * @param   string      $data   The data to be written
+	 *
+	 * @return  int|bool    Number of bytes written or FALSE on error
+	 */
+	public function stream_write( $data ) {
+		$length = fwrite( $this->file, $data );
+
+		if ( FALSE === $length ) {
+			trigger_error( 'Error writing to file: ' . $this->path,
+				E_USER_WARNING );
+			return FALSE;
+		}
+
+		// Upload to file service
+		$result = $this->client
+			->upload_file( $this->uri, $this->path );
+		if ( is_wp_error( $result ) || $result instanceof \WP_Error ) {
+			trigger_error( $result->get_error_message(), E_USER_WARNING );
+			return FALSE;
+		}
+
+		return $length;
+	}
+
+	/**
+	 * Delete a file
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 * @param   string  $path
+	 *
+	 * @return  bool    True if success. False on failure
+	 */
+	public function unlink( $path ) {
+		$path = $this->trim_path( $path );
+		$result = $this->client->delete_file( $path );
+
+		if ( is_wp_error( $result ) || $result instanceof \WP_Error ) {
+			trigger_error( $result->get_error_message(), E_USER_WARNING );
+			return FALSE;
+		}
+
+		$this->close_handler();
+
+		return TRUE;
+	}
+
+	/**
+	 * Get file stats
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 *
+	 * @return  array   The file statistics
+	 */
+	public function stream_stat() {
+		return fstat( $this->file );
+	}
+
+	/**
+	 * Get file stats by path
+	 *
+	 * Use by functions like is_dir, file_exists etc.
+	 * See: http://php.net/manual/en/streamwrapper.url-stat.php
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 *
+	 * @param   string      $path
+	 * @param   int         $flags
+	 *
+	 * @return  array|bool  The file statistics or false if failed
+	 */
+	public function url_stat( $path, $flags ) {
+		$path = $this->trim_path( $path );
+
+		// Default stats
+		$stats = array (
+			0         => 0,
+			'dev'     => 0,
+			1         => 0,
+			'ino'     => 0,
+			2         => 16895,
+			'mode'    => 16895,
+			3         => 0,
+			'nlink'   => 0,
+			4         => 0,
+			'uid'     => 0,
+			5         => 0,
+			'gid'     => 0,
+			6         => -1,
+			'rdev'    => -1,
+			7         => 0,
+			'size'    => 0,
+			8         => 0,
+			'atime'   => 0,
+			9         => 0,
+			'mtime'   => 0,
+			10        => 0,
+			'ctime'   => 0,
+			11        => -1,
+			'blksize' => -1,
+			12        => -1,
+			'blocks'  => -1,
+		);
+
+		$extension = pathinfo( $path, PATHINFO_EXTENSION );
+		/**
+		 * If the file is actually just a path to a directory
+		 * then return it as always existing. This is to work
+		 * around wp_upload_dir doing file_exists checks on
+		 * the uploads directory on every page load.
+		 *
+		 * Added by Joe Hoyle
+		 *
+		 * Hanif's note: Copied from humanmade's S3 plugin
```

Comment:
Let's add a link to the code on their GitHub repo.

Reply:
Done!